icu4c/source/i18n/calendar.cpp

```diff
@@ -1530,9 +1530,16 @@ void Calendar::computeGregorianFields(int32_t julianDay, UErrorCode& ec) {
         ec = U_ILLEGAL_ARGUMENT_ERROR;
         return;
     }
-    Grego::dayToFields(julianDay, fGregorianYear, fGregorianMonth,
-                       fGregorianDayOfMonth, gregorianDayOfWeekUnused,
-                       fGregorianDayOfYear, ec);
+    int32_t gregorianMonth, gregorianDayOfMonth, gregorianDayOfYear;
+    Grego::dayToFields(julianDay, fGregorianYear, gregorianMonth,
+                       gregorianDayOfMonth, gregorianDayOfWeekUnused,
+                       gregorianDayOfYear, ec);
+    U_ASSERT(gregorianMonth <= 12);
+    U_ASSERT(gregorianDayOfMonth <= 31);
+    U_ASSERT(gregorianDayOfYear <= 366);
```

Comment:
optional: Does Grego::dayToFields() not guarantee these? If the computations there don't necessarily guarantee these, then consider moving these asserts there.